### DISCUSSION TITLE
build error when run scripts/build_js.js

### DISCUSSION THREAD

**Discussion by shinnqy** [perspective-dev/perspective#2084] [Q&A] [answered]
## Support Question

Hi, I'm trying to run `scripts/build_js.js`. But it hit the error `'boost/multiprecision/cpp_int.hpp' file not found`. I already run `node scripts/install_tools.js`, so I cannot figure out what missed. Would you please help me how to resolve this?

![image](https://user-images.githubusercontent.com/15121888/212556165-98d2f66c-d53a-4213-894e-2201313e3b45.png)

### Follow up
I've resolved this by installing boost with another guide from google result.

But then I hit another error.
![image](https://user-images.githubusercontent.com/15121888/212559472-20315631-ea3b-42a6-a59f-aa9f9c106c7a.png)

**Comment by sinistersnare** [comment]
You should not be running build.js directly. Run `yarn` to install dependencies, then `yarn setup` and `yarn build` to setup and build.

**Comment by shinnqy** [comment]
The same. It will finally run `node build_js.js`
<img width="413" alt="image" src="https://user-images.githubusercontent.com/15121888/212690225-69de2bca-65e5-41ea-bedc-28f9de1da42d.png">

**Comment by texodus** [comment, marked as answer]
@shinnqy We can't debug your setup with screenshots.  This isn't an _issue_ with the project - [CI runs the build just fine e.g.](https://github.com/finos/perspective/actions/runs/3885276432) - even if it were, the bar for "Issues" is reproducibility, in your case, a GitHub repo which fails to build would work.

If I had to guess, it looks like you may have the [wrong emscripten version](https://github.com/finos/perspective/blob/master/package.json#L14) installed.  The build process will use the one in your environment rather than installing its own if it is detected.

Re: `boost` and the `install_tools` script, this is only configured to run on Linux systems (and your font looks like Menlo).  We can take this out now I think but this script does the same thing as `brew`.
